### ISSUE TITLE
The mAP after fine-turning and the train tricks 

### ISSUE THREAD

**Issue by RushDon** [chuanqi305/MobileNetv2-SSDLite#13]
Hi，
Thanks for you works firstly. I have some questions about this network.

First, the slice layers used for padding  [ 0 0 1 1] will incerase the time of forward-backward from 60ms to 90ms on my machine. By the caffe time tool, the increased time is not mainly from the slice layers, but  slice layers will cause other layers slower. do you have any experiences to show if this padding affects the mAP significantly? if not, it is not necessary to add additional layers. 

In order to verify the ReLU6，I replace the ReLU with ReLU6 in mobilenetv1-ssd，ReLU6 don't improve the mAP at all, but reduced by about 1%. how about the function of ReLU6 in V2? 

The mAP of the Mobilenet V2-SSDlite on VOC data can only reach about 20% without pre-trained model. how about your mAP afer fine-turning with transfered pre-model and without pre-model respectively? it seems that it is very hard to train this network. Could you please share some training skills?

**Comment by BalajiB197** [comment]
@RushDon 
Hi,
I am trying to train custom dataset. Newbie in caffe, can you help me with the steps to start training.
The usage in readme did'nt help me much.
Thanks,
Balaji

**Comment by RushDon** [comment]
@BalajiB197 
Hi, there is an instruction about  how to train on your own data.
https://github.com/chuanqi305/MobileNetv2-SSDLite#train-your-own-dataset
it should be very celar. if you have troubles with them, refer to 
https://github.com/chuanqi305/MobileNet-SSD#train-your-own-dataset
this should be helpful too.
actually the only thing you need to do is change the directory address correspongdingly.

**Comment by BalajiB197** [comment]
@RushDon 
I ran caffe_demo.py but my output is not proper other classes bounding boxes are added.
Actually I get error when I use /ssdlite/coco/deploy.prototxt with /ssdlite/deployprototxt.
But not facing any error when I run /ssdlite/deploy.prototxt with /ssdlite/deployprototxt and gives me wrong prediction. Don't mind if my question is silly.
![caffe_ssdlite](https://user-images.githubusercontent.com/22789396/40773843-af984fa6-64e1-11e8-88a9-a9570b9324aa.png)

**Comment by BalajiB197** [comment]
@RushDon 
Thanks for directing me in training.
But i am looking for step to create trainlmdb and testlmdb with small size custom datas.

**Comment by RushDon** [comment]
@BalajiB197 
I think the transfered model need to be fine-turned, it can not be applied directory because the details of the model  in tensorflow and the caffe are a little different.

**Comment by BalajiB197** [comment]
@RushDon
Thanks, I will try to rebuild and check. Have you try to run for ssdlite mobilenetv2 on top of caffe?
If you have a proper file can you share me(balajib197@gmail.com).

**Comment by dlyldxwl** [comment]
@BalajiB197 @RushDon  
hi, I have some problems to get mobilenet v2 ssdlite/ssd caffemodel, because my computer memory isn't enough.
can you share me Mobilenetv2 ssd_voc.caffemodel,  Mobilenetv2 ssdl_voc.caffemodel,  Mobilenetv2 ssdl_coco.caffemodel?
my email is 844518133@qq.com.
Thank you very much~

**Comment by BalajiB197** [comment]
@dlyldxwl 
I am looking for the same. Even I have issues with my machine for training.

**Comment by RushDon** [comment]
@dlyldxwl the trained models are terrible, which can not be applied. if you have the memeory problems, you can uncomment the engine: CAFFE to use the cpu to train the model. slow but work.

**Comment by BalajiB197** [comment]
@RushDon Same issue I am facing, where exactly we need to uncomment the caffe: engine.

**Comment by RushDon** [comment]
@BalajiB197 In the gen_model.py, search key words  engine, you will find that the engine:CAFFE is commented. remove the # to make it effective. Then search the batch_size for the train, replace the original value with a new value. 12 works for me.

**Comment by dlyldxwl** [comment]
@RushDon  Thank you for you reply!
my computer memory is 12G, when I run the load_caffe_weights file, it has memory error. 
could you send me  Mobilenetv2 ssd_voc.caffemodel, Mobilenetv2 ssdl_voc.caffemodel, Mobilenetv2 ssdl_coco.caffemodel  converted from TF.  I want to train the caffemodel.
my email is 844518133@qq.com.
thank you very much~

**Comment by RushDon** [comment]
@dlyldxwl I didn't tranfer them from TF. what i need is to train on my own data, there is no classes i need in the VOC and COCO data. So i need to find how to train from zero.

**Comment by BalajiB197** [comment]
@RushDon I found the caffe engine and uncomment it, but still facing memory issue.
No solution found for proper caffe model which runs ssdlite_mobilenetv2 ?

**Comment by RushDon** [comment]
@BalajiB197 how about to set the batch_size to 1?

**Comment by BalajiB197** [comment]
@RushDon I am also working on the same to train on my own images, can we connect in mail?
balajib197@gmail.com
